### PR TITLE
feat: Support USE CATALOG syntax and current_catalog() function

### PR DESCRIPTION
**Pull request by flashmouse**
I hereby agree to the terms of the CLA available at: https://docs.databend.com/dev/policies/cla/

## Summary
- fixes: #16848 

## Tests

- [x] Unit Test
- [ ] Logic Test
- [ ] Benchmark Test
- [ ] No Test - _Explain why_

## Type of change

- [ ] Bug Fix (non-breaking change which fixes an issue)
- [x] New Feature (non-breaking change which adds functionality)
- [ ] Breaking Change (fix or feature that could cause existing functionality not to work as expected)
- [ ] Documentation Update
- [ ] Refactoring
- [ ] Performance Improvement
- [ ] Other (please describe):

- - -
This change is [Reviewable](https://reviewable.io/reviews/databendlabs/databend/16926)

### CODE REVIEW

**Comment by flashmouse** [comment]
I observe that `use_database` statement is prior to `use_catalog` unless I quote #ident(catalog)
```
use catalog abc  // parse to UseDatabase{"catalog "} which is unexpected
use catalog `abc` //  parse to UseCatalog{"abc"} correctly
```

File: src/query/ast/src/parser/statement.rs [hunk 1]
```diff
@@ -516,6 +516,12 @@ pub fn statement_body(i: Input) -> IResult<Statement> {
             })
         },
     );
+    let use_catalog = map(
+        rule! {
+            (USE ~ CATALOG) ~ #ident
+        },
```
Comment (b41sh) [inline src/query/ast/src/parser/statement.rs:522]: remove the parenthesis will fix the matched as `use database` issues
```rust
        rule! {
            USE ~ CATALOG ~ #ident
        },
```

Comment (flashmouse) [reply src/query/ast/src/parser/statement.rs:522]: I removed the parenthesis but still the same result

Comment (b41sh) [reply src/query/ast/src/parser/statement.rs:522]: I made a mistake. It's okay to keep the parenthesis.
This problem should be caused by the MySQL client, you can use the [BendSQL](https://github.com/databendlabs/bendsql) client to test this case. Add please add some test cases to this file `src/query/ast/tests/it/parser.rs` for verification.

**Comment by flashmouse** [comment]
@b41sh  please help me check if it is suitable now

**Comment by b41sh** [comment]
Hi @flashmouse  Is this PR ready to review? Seems almost OK for me. Better to add some test cases in `tests/suites/3_stateful_iceberg/00_rest/00_0000_create_and_show.sh` and `tests/suites/3_stateful_iceberg/00_rest/00_0000_create_and_show.result`

**Comment by flashmouse** [comment]
> Hi @flashmouse Is this PR ready to review? Seems almost OK for me. Better to add some test cases in `tests/suites/3_stateful_iceberg/00_rest/00_0000_create_and_show.sh` and `tests/suites/3_stateful_iceberg/00_rest/00_0000_create_and_show.result`

I would mark this as **ready to review** after add test as you advice.

File: tests/suites/3_stateful_iceberg/00_rest/00_0000_create_and_show.sh [hunk 2]
```diff
@@ -20,4 +20,10 @@ echo "SHOW CREATE CATALOG iceberg_ctl;" | $BENDSQL_CLIENT_CONNECT
 
 echo "SHOW DATABASES IN iceberg_ctl;" | $BENDSQL_CLIENT_CONNECT
 
+echo "SELECT CURRENT_CATALOG();" | $BENDSQL_CLIENT_CONNECT
+
+echo "USE CATALOG `iceberg_ctl`;" | $BENDSQL_CLIENT_CONNECT
```
Comment (b41sh) [inline tests/suites/3_stateful_iceberg/00_rest/00_0000_create_and_show.sh:25]: Remove quotes, write as `USE CATALOG iceberg_ctl;` will fix the test.

Comment (flashmouse) [reply tests/suites/3_stateful_iceberg/00_rest/00_0000_create_and_show.sh]: each call ``BENDSQL_CLIENT_CONNECT="bendsql -uroot --host ${QUERY_MYSQL_HANDLER_HOST} --port ${QUERY_HTTP_HANDLER_PORT} --database ${MYSQL_DATABASE} --quote-style=never"`` in this test would create a new connection cause reset session, I think it's the reason this test fail.

And now this should be fixed now .

**Review by b41sh — APPROVED** [review]
@flashmouse LGTM, thanks for your contribution.

**Pull request merged**